urdf/wheel.urdf.xacro

```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0" encoding="utf-8"?>
+<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
+
+  <!-- wheel defining macro -->
+  <xacro:macro name="wheel" params="wheel_props prefix">
+
+    <xacro:property name="wheel_separation_x" value="0.44" />
+
+    <xacro:property name="rpy" value="0 0 0" />
+    <xacro:if value="${prefix == 'front_left'}">
+      <xacro:property name="x" value="${wheel_separation_x/2.0}" />
+      <xacro:property name="y" value="${wheel_props['wheel_separation']/2.0}" />
+      <xacro:property name="inertia_y_offset" value="${wheel_props['inertia_y_offset']}" />
+      <xacro:property name="mesh"
+        value="file://$(find ${wheel_props['mesh_package']})/${wheel_props['folder_path']}/wheel_a.dae" />
+    </xacro:if>
+    <xacro:if value="${prefix == 'front_right'}">
+      <xacro:property name="x" value="${wheel_separation_x/2.0}" />
+      <xacro:property name="y" value="-${wheel_props['wheel_separation']/2.0}" />
+      <xacro:property name="inertia_y_offset" value="${-wheel_props['inertia_y_offset']}" />
+      <xacro:property name="mesh"
+        value="file://$(find ${wheel_props['mesh_package']})/${wheel_props['folder_path']}/wheel_b.dae" />
+    </xacro:if>
+    <xacro:if value="${prefix == 'rear_left'}">
+      <xacro:property name="x" value="-${wheel_separation_x/2.0}" />
+      <xacro:property name="y" value="${wheel_props['wheel_separation']/2.0}" />
+      <xacro:property name="inertia_y_offset" value="${wheel_props['inertia_y_offset']}" />
+      <xacro:property name="mesh"
+        value="file://$(find ${wheel_props['mesh_package']})/${wheel_props['folder_path']}/wheel_c.dae" />
+    </xacro:if>
+    <xacro:if value="${prefix == 'rear_right'}">
+      <xacro:property name="x" value="-${wheel_separation_x/2.0}" />
+      <xacro:property name="y" value="-${wheel_props['wheel_separation']/2.0}" />
+      <xacro:property name="inertia_y_offset" value="${-wheel_props['inertia_y_offset']}" />
+      <xacro:property name="mesh"
+        value="file://$(find ${wheel_props['mesh_package']})/${wheel_props['folder_path']}/wheel_d.dae" />
+    </xacro:if>
```

Comment:
I think that it will be better to pass these properties as arguments and define them in `panther_macro`.

Reply:
I wanted to keep adding and parametrizing the wheel entirely within the internal macro. I would wait for @pawelirh for his opinion on this manner.

Reply:
Fun fact, I followed exactly the same pattern as @macstepien proposed in shuttle_car_macro.urdf.xacro [LINK](https://github.com/hsrn24/shuttle_car/blob/task/create_shuttle_car_URDF/shuttle_car_description/models/shuttle_car_base/shuttle_car_macro.urdf.xacro) (Don't know if you have access).

Anyway, it's the vehicle's business how wide to set the wheels, not the wheels' business. I pass the parameters `wheel_separation_x` and `wheel_separation_y` from the macro to wheels definition.